### PR TITLE
Add passport scan tutorial

### PR DESCRIPTION
**Pull request by IDima1**

### CODE REVIEW

File: Rarime/Code/UI/Views/StepIndicator.swift [hunk 1]
```diff
@@ -0,0 +1,27 @@
+//
+//  StepIndicator.swift
+//  Rarime
+//
+//  Created by Dmytro Ivaniv on 03.09.2024.
+//
```
Comment (ardier16) [inline Rarime/Code/UI/Views/StepIndicator.swift:6]: plz remove theses comments, we're not using them

File: Rarime/Code/Modules/ScanPassport/Enums/PassportScanStep.swift [hunk 2]
```diff
@@ -1,7 +1,13 @@
 import Foundation
 
 enum PassportScanStep: Int, CaseIterable {
```
Comment (ardier16) [inline Rarime/Code/Modules/ScanPassport/Enums/PassportScanStep.swift:3]: PassportTutorialStep

File: Rarime/Code/Modules/ScanPassport/Views/PassportScanTutorialView.swift [hunk 3]
```diff
@@ -63,41 +60,32 @@ struct PassportScanTutorialView: View {
...
 
 private struct PassportScanTutorialStep: View {
-    let title: LocalizedStringResource
-    let description: LocalizedStringResource
-    let videoURL: URL
-    let actionText: LocalizedStringResource
+    let step: PassportScanStep
     let action: () -> Void
-    @Binding var currentStep: Int
+    @Binding var currentStep: PassportScanStep
```
Comment (ardier16) [inline Rarime/Code/Modules/ScanPassport/Views/PassportScanTutorialView.swift:65]: why changed to enum object binding?

File: Rarime/Code/Constants/Videos.swift [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import Foundation
+
+enum Videos {
+    static let removeCase = Bundle.main.url(forResource: "RemoveCase", withExtension: "mp4")!
+    static let scanMrzPassport = Bundle.main.url(forResource: "ScanMrzPassport", withExtension: "mp4")!
+    static let readNfcPassport = Bundle.main.url(forResource: "ReadNfcPassport", withExtension: "mp4")!
```
Comment (ardier16) [inline Rarime/Code/Constants/Videos.swift:6]: let's remove "Passport" suffix, it was needed to differ between id card and external passport

File: Rarime/Code/Modules/ScanPassport/Enums/PassportTutorialStep.swift [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        case .removeCase: return Videos.removeCase
+        case .scanYourPassport: return Videos.scanMrzPassport
+        case .readingPassportData: return Videos.readNfcPassport
+        }
+    }
+    
+    var buttonText: LocalizedStringResource {
+        switch self {
+        case .removeCase: return "Next"
+        case .scanYourPassport: return "Next"
```
Comment (ardier16) [inline Rarime/Code/Modules/ScanPassport/Enums/PassportTutorialStep.swift:33]: case .removeCase, . scanYourPassport

File: Rarime/Code/Modules/ScanPassport/Enums/PassportTutorialStep.swift [hunk 6]
```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+enum PassportTutorialStep: Int, CaseIterable {
+    case removeCase, scanYourPassport, readingPassportData
```
Comment (ardier16) [inline Rarime/Code/Modules/ScanPassport/Enums/PassportTutorialStep.swift:4]: - removeCase
- scanMrz
- readNfc

File: Rarime/Code/Modules/ScanPassport/Enums/PassportTutorialStep.swift [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    
+    var title: LocalizedStringResource {
+        switch self {
+        case .removeCase: return "Remove case"
```
Comment (ardier16) [inline Rarime/Code/Modules/ScanPassport/Enums/PassportTutorialStep.swift:8]: "Remove Case"

File: Rarime/Code/Modules/ScanPassport/Enums/PassportTutorialStep.swift [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        case .scanYourPassport: return "Scan your Passport"
```
Comment (ardier16) [inline Rarime/Code/Modules/ScanPassport/Enums/PassportTutorialStep.swift:9]: "Scan Your Passport"

File: Rarime/Code/Modules/Home/Views/HomeView.swift [hunk 9]
```diff
@@ -23,6 +23,9 @@ struct HomeView: View {
     @State private var isBalanceFetching = true
     @State private var pointsBalance: PointsBalanceRaw? = nil
     @State private var cancelables: [Task<Void, Never>] = []
+    
+    @State private var isTutorialPresented = false
+    @State private var isTutorialShown = AppUserDefaults.shared.isScanTutorialDisplayed
```
Comment (ardier16) [inline Rarime/Code/Modules/Home/Views/HomeView.swift:28]: move it to scan passport view

File: Rarime/Code/Modules/Home/Views/HomeView.swift [hunk 10]
```diff
@@ -160,12 +163,17 @@ struct HomeView: View {
                 }
             )
         )
-        .sheet(isPresented: $isCreateIdentityIntroPresented) {
+        .sheet(isPresented: $isCreateIdentityIntroPresented, onDismiss: {
+            presentScanTutorialIfNeeded()
```
Comment (ardier16) [inline Rarime/Code/Modules/Home/Views/HomeView.swift:167]: will not be needed here

File: Rarime/Code/Constants/Videos.swift [hunk 11]
```diff
@@ -2,6 +2,6 @@ import Foundation
 
 enum Videos {
     static let removeCase = Bundle.main.url(forResource: "RemoveCase", withExtension: "mp4")!
-    static let scanMrzPassport = Bundle.main.url(forResource: "ScanMrzPassport", withExtension: "mp4")!
-    static let readNfcPassport = Bundle.main.url(forResource: "ReadNfcPassport", withExtension: "mp4")!
+    static let scanMrz = Bundle.main.url(forResource: "ScanMrzPassport", withExtension: "mp4")!
```
Comment (ardier16) [inline Rarime/Code/Constants/Videos.swift:5]: plz rename underlying video files as well to always match with the enum keys

File: Rarime/Code/Modules/Home/Views/HomeView.swift [hunk 12]
```diff
@@ -19,13 +19,11 @@ struct HomeView: View {
 
     @State private var isCongratsShown = false
     @State private var isClaimed = false
+    @State private var shouldShowTutorial = false
```
Comment (ardier16) [inline Rarime/Code/Modules/Home/Views/HomeView.swift:22]: seems like unused

**Review by ardier16 — APPROVED** [review]

**Review by ivanlele — APPROVED** [review]

**Pull request merged**